scripts/detailed_log.py

```diff
@@ -8,35 +8,42 @@
 log_path = os.path.join(base_dir, 'logs')
 file_list = os.listdir(log_path)
 
-# Modify this list with whatever bosses you want to look at.
 bosses = ['articuno', 'zapdos', 'moltres', 'mewtwo', 'raikou', 'entei',
           'suicune', 'lugia', 'ho-oh', 'latias', 'latios', 'kyogre', 'groudon',
           'rayquaza', 'uxie', 'mesprit', 'azelf', 'dialga', 'palkia', 'heatran',
-          'giratina-altered', 'cresselia', 'tornadus', 'thundurus', 'reshiram', 'zekrom',
-          'landorus', 'kyurem', 'xerneas', 'yveltal', 'zygarde-50', 'tapu-koko', 'tapu-lele',
-          'tapu-bulu', 'tapu-fini', 'solgaleo', 'lunala', 'necrozma', 'nihilego', 'xurkitree',
-          'buzzwole', 'pheromosa', 'celesteela', 'kartana', 'guzzlord', 'stakataka',
-          'blacephalon'
+          'giratina-altered', 'cresselia', 'tornadus-incarnate', 'thundurus-incarnate',
+          'reshiram', 'zekrom', 'landorus-incarnate', 'kyurem', 'xerneas', 'yveltal',
+          'zygarde-50', 'tapu-koko', 'tapu-lele', 'tapu-bulu', 'tapu-fini', 'solgaleo',
+          'lunala', 'necrozma', 'nihilego', 'xurkitree', 'buzzwole', 'pheromosa',
+          'celesteela', 'kartana', 'guzzlord', 'stakataka', 'blacephalon'
           ]
 global_losses = 0
 global_wins = 0
 global_shinies = 0
 global_legends = 0
+boss_files = {}
 
 for boss in bosses:
+    found_files = []
+    for x in file_list:
+        if boss.lower() in x.lower() and ('.log' in x or '.txt' in x):
+            # add to the files found
+            found_files.append(x)
+            # remove from the original list since we're done with it
+            file_list.remove(x)
```

Comment:
Your error is due to this. I'm gonna check but usually removing objects from a list while iterating over it leads to issues. You don't need to remove it from the list anyway